cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -623,8 +623,8 @@ func EncodePublicKeyPEM(key crypto.PublicKey) ([]byte, error) {
 // NewPrivateKey returns a new private key.
 var NewPrivateKey = GeneratePrivateKey
 
-func GeneratePrivateKey(keyType x509.PublicKeyAlgorithm) (crypto.Signer, error) {
-	if keyType == x509.ECDSA {
+func GeneratePrivateKey(keyType kubeadmapi.EncryptionAlgorithmType) (crypto.Signer, error) {
+	if keyType == kubeadmapi.EncryptionAlgorithmECDSA {
 		return ecdsa.GenerateKey(elliptic.P256(), cryptorand.Reader)
```

Comment:
Just curious, why it's P256? any reference for this?

Reply:
it was a recommendation by SIG Auth IIRC, who reviewed the KEP.
our proposal to them was to use AES-GCM, symmetric encryption of minimum key size 32 bytes.
that discussion was somewhere in k/enhancements.


P-256 provides the 128bits which is what we wanted and it's a go to option for ECDSA.

> the NIST Curve P-256 which provides 128-bits of security

https://csrc.nist.gov/csrc/media/events/workshop-on-elliptic-curve-cryptography-standards/documents/papers/session6-adalier-mehmet.pdf